### DISCUSSION TITLE
Make it optional to reverse sort order of stacked items

### DISCUSSION THREAD

**Discussion by kustodian** [grafana/grafana#43538] [Dashboards, Panels & Visualizations]
In Grafana when we have e.g. 3 queries A, B, C in the order like that from top to bottom if we stack those queries will be stacked in that order as well: A bottom, B on top of it and C on top. This is not that important until we start changing the order in the Legend. If we set the order in the legend to "Decreasing", the values on the graph will be stacked from highest to lowest value, e.g. like this:

![image](https://user-images.githubusercontent.com/1949229/147468941-57186588-8365-4059-b3f8-d63c472e88df.png)

This means that the Legend on top will have the highest value, while on the graph the highest value will be on the bottom. When I see this I always have to stop for a moment to process that they are in a different order. I guess is useful often, but on many occasions, I find it more user friendly that the values on the Legend are sorted the same as on the graph, e.g. like this:

![image](https://user-images.githubusercontent.com/1949229/147468984-38d6d024-e969-464e-bb2f-17005fcd2342.png)

This becomes even more important when we start displaying a Tooltip with all metrics sorted also in a Decreasing order. Then it's even harder to follow which metric has which value. Check this one out:

![image](https://user-images.githubusercontent.com/1949229/147469102-fdb05dbe-d916-49df-8a80-9e7a50b57167.png)

Legend and Tooltip are sorted decreasing, while on the graph it's displayed in the reverse order.

Having an option like: `Stack Order: Reversed/Normal` would solve this problem.

**Comment by marco-kusa** [comment]
bump, what was the reasoning when this was set inverted like this? Can it be changed?

**Comment by hatamiarash7** [comment]
I have something like this:
<img width="1019" alt="image" src="https://github.com/grafana/grafana/assets/16325641/18fe65c5-0c06-49e9-92c4-5a90a9604725">

It's not good !!! It should be like this:
<img width="1017" alt="image" src="https://github.com/grafana/grafana/assets/16325641/49f1a3e0-a076-4182-86fc-008c7034be23">

we need to reverse the order

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
